styles/Canonical/009-headings-no-links.yml

```diff
@@ -0,0 +1,16 @@
+# 09 - Heading - Headings should not contain links
+
+extends: existence
+message: "Headings should not contain links"
+link: https://docs.ubuntu.com/styleguide/en/#other-considerations
+nonword: true
+level: warning
+scope: raw
+tokens:
+  - '[\#]+ [ \w]+\[[\w]+[ \w]*\]\([.\w]+\)'
+  - '[ \w]+\[[\w]+[ \w]*\]\([.\w]+\)[^\n]*\n[=-]{2,}\n'
+  - '[\#]+ [ \w]+<http[^\n]+>'
+  - '[ \w]+<http[^\n]+>[^\n]*\n[=-]{2,}\n'
+  - '[ \w]+`[ \w]*<[\.a-zA-Z0-9\/]+>`_\n(.)\1{3,}\n'
+  - '[ \w]+\:ref\:`[^\n]+`[ \w]*\n(.)\2{3,}\n'
+  - '[ \w]+\:doc\:`[^\n]+`[ \w]*\n(.)\3{3,}\n'
```

Comment:
You could add the equivalent for `{ref}` and `{doc}` (which is what MyST uses instead of `:ref:` and `:doc:`).